src/MongoDB.Driver.Core/Core/Servers/Server.cs

```diff
@@ -46,8 +46,6 @@ internal sealed class Server : IClusterableServer
         // static fields
         private static readonly List<Type> __invalidatingExceptions = new List<Type>
         {
-            typeof(MongoNotPrimaryException),
-            typeof(MongoNodeIsRecoveringException),
```

Comment:
`MongoNotPrimaryException` and `MongoNodeIsRecoveringException` removed because we no longer can immediately invalidate a server when we see these exception types: we need to apply the staleness check logic.